### ISSUE TITLE
USSD PUSH configuration issues with SS7-SIM

### ISSUE THREAD

**Issue by armachea** [RestComm/ussdgateway#93]
**What happened**:

Hi guys, hope everyone is having a great day so far. I have restcomm-ussd-7.1.0-SNAPSHOT--wildfly-10.1.0.Final installed on Ubuntu [16.04] I have tested all scenarios but when it comes to ussd push am getting an error about local SSN. 

PostMan <is sending an XML payload push> 
                                            >>>> USSD GW <Received payload and send SRI-request>
                                                                                >>>> SS7-SIM <Received SRI-request then STOP>
**What you expected to happen**:

Well I am pretty sure, my SS7-SIM should receive more than the SRI req/resp. It supposed to received the string message from the xml payload. But i can't receive the message.

**How to reproduce it (as minimally and precisely as possible)**:
This is what i get from USSD GW after firing Postman payload.
WARN FROM USSD GW:
||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||
13:01:25,404 WARN  [org.mobicents.protocols.ss7.sccp.impl.SccpRoutingControl] (pool-5-thread-1) Received SccpMessage=Sccp Msg [Type=UDT networkId=0 sls=6 incomingOpc=2 incomingDpc=1 outgoingDpc=-1 CallingAddress(pc=2,ssn=6,AI=83,gt=GlobalTitle0100Impl [digits=000, natureOfAddress=INTERNATIONAL, numberingPlan=ISDN_TELEPHONY, translationType=0, encodingScheme=BCDOddEncodingScheme[type=BCD_ODD, code=1]]) CalledParty(pc=0,ssn=8,AI=18,gt=GlobalTitle0100Impl [digits=923330053058, natureOfAddress=INTERNATIONAL, numberingPlan=ISDN_TELEPHONY, translationType=0, encodingScheme=BCDEvenEncodingScheme[type=BCD_EVEN, code=2]]) DataLen=77] for Translation but no local SSN is present for primary Address
||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||

**Anything else we need to know?**:

PostMan is running on : 192.168.43.12
USSD-GW is running on: 192.168.43.13
SS7-SIM is runnin on: 192.168.43.14
I have restcomm-ussd-7.1.0-SNAPSHOT running on wildfly 10.1.0

**Comment by nhanth87** [comment]
I think you already change the SSN number to another number (example 147) instead of 8?

Br,
TN

**Comment by armachea** [comment]
Yes, That's correct i tried with 147 and 8 for local SSN........ unfortunately, i got the same WARN for both case.

nhanth87, is there a thing am missing? I am totally out of solution.

**Comment by armachea** [comment]
Sure thing, this is the .xml files for ussd-gw.
[xmlFILES.zip](https://github.com/RestComm/ussdgateway/files/2183868/xmlFILES.zip)

**Comment by armachea** [comment]
Guys, any advice regarding this problem.......

**Comment by armachea** [comment]
Hi nhanth87,

I just wanna thank you for the time and guidance you giving me, I have added the above config but something is off. am receiving a warning that i am sending a ussd push FROM PC=0 TO PC=0. This is crazy, because i have updated the routing address in standalone.xml and  SccpStack_sccprouter2.xml in USSD GW.

||||||||||||| USSD PUSH INSIDE POSTMAN ||||||||||||||||
<?xml version="1.0" encoding="UTF-8" ?>
<dialog mapMessagesSize="1">
<unstructuredSSNotify_Request dataCodingScheme="15" string="Your new balance is 34.38 AFN and expires on 30.07.2012. Cost of last event was 0.50 AFN.">
<msisdn number="923330053099" nai="international_number" npi="ISDN"/>
</unstructuredSSNotify_Request>
</dialog>

||||||||||||| UPDATED STANDALONE.XML USSD GW ||||||||||||||||
<mbean name="TcapStack" type="TcapStack" class="org.mobicents.protocols.ss7.tcap.TCAPStackImpl" enabled="true" reflection="false">
<property name="sccpStack" type="Bean" value="SccpStack"/>
<property name="ssn" type="int" value="8"/>
<property name="previewMode" type="bool" value="false"/> 
<property name="extraSsns" type="Map" value="empty">
<entry key="1" type="int" value="6"/>
</property>               
</mbean>

||||||||||||| UPDATED SccpStack_sccprouter2.xml USSD GW ||||||||||||||||
<routingAddress>
<id value="1"/>
<sccpAddress pc="2" ssn="8">
<ai value="19"/>
<gt type="GT0100" tt="0" es="1" np="1" nai="4" digits="*"/>
</sccpAddress>
<id value="2"/>
<sccpAddress pc="1" ssn="6">
<ai value="19"/>
<gt type="GT0100" tt="0" es="1" np="1" nai="4" digits="*"/>
</sccpAddress>
</routingAddress>


????? Not to forget that i have update the imsi number with random digit from ss7-sim side because without the imsi number was receiving an error regarding [TypeMismatch]......... All of the changes i have done is leading me to | ussd push FROM PC=0 TO PC=0 | with ss7-sim only receiving SRI req/res.

This is the WARN
==============
13:22:15,750 WARN  [org.mobicents.protocols.ss7.sccp.impl.SccpRoutingControl] (SLEE-EventRouterExecutor-3-thread-1) Received SccpMessage for Translation but no matching Rule found for local routing
SccpMessage=Sccp Msg [Type=-1 networkId=0 sls=2 incomingOpc=-1 incomingDpc=-1 outgoingDpc=-1
CallingAddress(pc=0,ssn=8,AI=18,gt=GlobalTitle0100Impl [digits=923330053058, natureOfAddress=INTERNATIONAL,numberingPlan=ISDN_TELEPHONY, translationType=0, encodingScheme=BCDEvenEncodingScheme[type=BCD_EVEN, code=2]])
CalledParty(pc=0,ssn=8,AI=18,gt=GlobalTitle0100Impl [digits=, natureOfAddress=INTERNATIONAL, numberingPlan=ISDN_TELEPHONY, translationType=0, encodingScheme=BCDEvenEncodingScheme[type=BCD_EVEN, code=2]]) DataLen=178]
==============

**Comment by armachea** [comment]
![updated sccpstack_sccprouter2 xml ussd gw](https://user-images.githubusercontent.com/25345092/42812950-e7b910fe-89c7-11e8-96bc-7d8cfe9d079b.jpg)
![updated standalone xml ussd gw](https://user-images.githubusercontent.com/25345092/42812951-e7d8db32-89c7-11e8-85b7-5045cbb16e95.jpg)
![ussd push from postman](https://user-images.githubusercontent.com/25345092/42812952-e7ff131a-89c7-11e8-8a55-6085d9d5df60.jpg)

**Comment by nhanth87** [comment]
The log looks good,
We now pass the issue `no local SSN is present for primary Address` and it's come to `no matching Rule found for local routing`

You need to look into the called party for the routing rule.

**Comment by nhanth87** [comment]
can you send full server.log, need to check it's SRIforSM or USS_Notify message?

**Comment by armachea** [comment]
It's definitely a SRIforSM....  

Please this is the full server.
[server.log](https://github.com/RestComm/ussdgateway/files/2201343/server.log)
log

**Comment by armachea** [comment]
Hi nhanth87,

Hope you doing fine.... I was wondering  if there is any changes needed from the SS7-SIM ? in order to make this work.

Regards

**Comment by armachea** [comment]
Hi guys Anybody here ?
